### PR TITLE
SI-8065 Regression in implicit resolution

### PR DESCRIPTION
**Pull request by rjolly**
Follow-up to https://github.com/scala/scala/pull/3540

### CODE REVIEW

File: src/compiler/scala/tools/nsc/typechecker/Implicits.scala [hunk 1]
```diff
@@ -847,6 +847,8 @@ trait Implicits {
           } else search
       }
 
+      private final val sortMatches = sys.props contains "scalac.debug.sortEligibleImplicits"
```
Comment (retronym) [inline src/compiler/scala/tools/nsc/typechecker/Implicits.scala:850]: Sorry I wan't clearer. I'd need to the default to be the same as 2.10 behaviour.

**Comment by retronym** [comment]
We also need some context in the commit message.

The guidelines are listed here: https://github.com/scala/scala/blob/master/CONTRIBUTING.md

**Comment by rjolly** [comment]
Jason,
I have worked around the problem in my project.
I do not need the system property.
The problem is that the compiler is indeterministic in its present state.
Is it worth the ~1% performance improvement ?

**Comment by retronym** [comment]
The risk of any change to implicit search is too high a week before 2.11.0-RC1. Your patch seems to suggest that changing it required a change to the Scalacheck test, which means that people are probably depending on the order, even though it is hard to predict.

I would like to get to the bottom of that, but it will have to be in 2.12.

**Comment by retronym** [comment]
BTW, I do apologise that we haven't been able to accept this as I know you've done a lot of work to help performance test the lack of sorting and verify that it works in your code base. SI-3346 was just such an important improvement for other aspects of implicit search, and its really just unfortunate that it clashed so hard with your use of views.

**Comment by rjolly** [comment]
Yes, there seem to be a problem with this scalacheck test, so the change is not 100% safe (by not much !) Ok, I guess 2.12 will be fine. Thank you for your insight and responsiveness.

**Pull request closed without merging**